src/include/buffer.h

```diff
@@ -382,13 +384,39 @@ namespace buffer CEPH_BUFFER_API {
   };
 
 
+  class hangable_ptr : public ptr, public boost::intrusive::list_base_hook<> {
```

Comment:
this needs a better name, forgive me for complaining

Reply:
maybe ptr_node or ptr_entry

Reply:
That's right, the current naming is just terrible. Guys, what do you think about `ptr_combined` per analogy to `raw_combined`? This would reflect combining `ptr` with the container's hooks. Then gluing `ptr_combined` with `raw` (keeping in mind that `raw_combined` is the typical case) would be *hypercombining* performed by `::create_hypercombined` we currently have.

I'm also perfectly fine with `ptr_entry` as it emphasize the intention of storing a `ptr` in the `bufferlist` container.

Reply:
I think ptr_entry/node is clearer

Reply:
if i understand correctly, every ptr has to be hangable in order to live in a bufferlist - so not all of these hangable_ptrs are combined

Reply:
@cbodley: yes, `ptr_node` fits much better. Just pushed amended version.